docs/hello_nf-core/03_add_module.md

```diff
@@ -1,63 +1,893 @@
+In this third part of the Hello nf-core training course, we show you how to add an existing nf-core module to your pipeline and adapt your local modules to follow nf-core conventions.
 
-TODO: THIS IS A WIP AND SHOULD NOT BE ADDED TO THE NAVIGATION
+One of the great advantages of nf-core pipelines is the ability to leverage pre-built, tested modules from the nf-core/modules repository. Rather than writing every process from scratch, you can install and use community-maintained modules that follow best practices.
 
-(NOT SUBJECT TO REVIEW)
+In this section, we'll replace the custom `collectGreetings` module with the `cat/cat` module from nf-core/modules, then progressively adapt our `cowpy` module to follow nf-core patterns.
+
+!!! note
+
+    This section assumes you have completed [Part 2: Rewrite Hello for nf-core](./02_rewrite_hello.md) and have a working `core-hello` pipeline.
+
+    If you didn't complete Part 2 or want to start fresh for this section, you can use the `core-hello-part2` solution as your starting point:
+
+    ```bash
+    cp -r hello-nf-core/solutions/core-hello-part2 core-hello
+    cd core-hello
+    ```
+
+    This gives you a fully functional nf-core pipeline ready for adding modules.
 
 ---
 
-## 1. Find cat/cat to replace collectGreetings
+## 1. Use an nf-core module
+
+First, let's learn how to find, install, and use an existing nf-core module in our pipeline.
+
+The `collectGreetings` process in our pipeline uses the Unix `cat` command to concatenate multiple greeting files into one. This is a perfect use case for the nf-core `cat/cat` module, which is designed specifically for concatenating files.
+
+!!! note "Module naming convention"
+
+    nf-core modules follow the naming convention `software/command`. The `cat/cat` module wraps the `cat` command from the `cat` software package. Other examples include `fastqc/fastqc` (FastQC software, fastqc command) or `samtools/view` (samtools software, view command).
+
+### 1.1. Browse available modules on the nf-core website
+
+The nf-core project maintains a centralized catalog of modules at [https://nf-co.re/modules](https://nf-co.re/modules).
+
+Navigate to the modules page in your web browser and use the search bar to search for "cat_cat".
+
+You should see `cat/cat` in the search results. Click on it to view the module documentation.
+
+The module page shows:
+
+- A description: "A module for concatenation of gzipped or uncompressed files"
+- Installation command: `nf-core modules install cat/cat`
+- Input and output channel structure
+- Available parameters
+
+### 1.2. List available modules from the command line
+
+You can also search for modules directly from the command line using nf-core tools.
+
+```bash
+nf-core modules list remote
+```
+
+This will display a list of all available modules in the nf-core/modules repository. You can scroll through or pipe to `grep` to find specific modules:
+
+```bash
+nf-core modules list remote | grep cat
+```
```

Comment:
ths list every string for me with cat in it 
<img width="569" height="408" alt="Screenshot 2025-10-21 at 11 53 37" src="https://github.com/user-attachments/assets/1f01ce67-fc2f-42bf-89f3-d319f0282d88" />